thanos/playbooks/ThanosCompactHalted.md

```diff
@@ -0,0 +1,162 @@
+    level=info ts=2023-04-27T05:43:18.591951424Z caller=intrumentation.go:81 msg="changing probe status" status=not-healthy reason="could not sync metas: filter metas: filter blocks marked for deletion: get file: 01GYH4WNQKAXY1Y4RN1K5J1R8Q/deletion-mark.json: open object: Timeout when reading or writing data"
+    level=info ts=2023-04-27T05:43:19.711718378Z caller=compact.go:1070 group="0@{cluster=\"s-eu-de-1\", cluster_type=\"scaleout\", prometheus=\"vmware-monitoring/vmware-vc-mgmt-b-0\", prometheus_replica=\"prometheus-vmware-vc-mgmt-b-0-0\", region=\"eu-de-1\"}" groupKey=0@2754565673212689501 msg="downloaded and verified blocks; compacting blocks" plan="[/data/compact/0@2754565673212689501/01GYZEZNMR42383F8BXZHE91A0 /data/compact/0@2754565673212689501/01GYZNVCWRNMVJZYY7MZD4SJQM /data/compact/0@2754565673212689501/01GYZWQ44S29Y97MV0CKBEVKRA /data/compact/0@2754565673212689501/01GZ03JVCRRQ9E6HAQ9333P9SY]" duration=1.12004828s duration_ms=1120
+    level=error ts=2023-04-27T05:43:19.742219509Z caller=compact.go:488 msg="critical error detected; halting" err="compaction: group 0@2754565673212689501: compact blocks [/data/compact/0@2754565673212689501/01GYZEZNMR42383F8BXZHE91A0 /data/compact/0@2754565673212689501/01GYZNVCWRNMVJZYY7MZD4SJQM /data/compact/0@2754565673212689501/01GYZWQ44S29Y97MV0CKBEVKRA /data/compact/0@2754565673212689501/01GZ03JVCRRQ9E6HAQ9333P9SY]: 2 errors: populate block: context canceled; context canceled"
+    ```
+
+2. Kick the pod.
+
+3. Check the log again like in in step `1.`.
+
+## Chunk critical error
+
+### Problem
+
+Some block error prohibits Thanos from continuing to compact. The faulty block needs to be identified and removed in your object storage (Swift, S3, Ceph, ...).
+
+### Solution
+
+#### 1. Check the logs of the Thanos compcator in question
+
+    ```bash
+    kubectl logs --follow $podName
+    ```
+
+    Usually you would see some critical error like this:
+    ```
+    level=error ts=2023-11-09T12:23:35.120966651Z caller=compact.go:487 msg="critical error detected; halting" err="compaction: group 0@7529044506654606473: compact blocks [/data/compact/0@7529044506654606473/01HE8PG6C507J89N60T3SKQN3S /data/compact/0@7529044506654606473/01HE8XBXM3AP4WDCS25DKJ6W4J /data/compact/0@7529044506654606473/01HE947MW4JBDY3648JSVYKPAW /data/compact/0@7529044506654606473/01HE9B3C499RNJ4SM5V8EKM0ZG]: populate block: chunk iter: cannot populate chunk 8 from block 01HE9B3C499RNJ4SM5V8EKM0ZG: segment index 0 out of range"
+
+    ```
+
+    If you can't see it, kick the pod and watch the logs straight, to catch the initial error. It might be obfuscated by appending messages by the time you are looking at it.
+
+#### 2. Usually the block in question has no content, in particular it is missing the `chunks` folder. 
+    Find out the compactor in question:
+
+    ```
+    kubectl get objectstore
+    # Example output
+    # NAME                AGE
+    # thanos-kubernetes   84d
+    ```
```

Comment:
This is specific to the Thanos operator, the respective config needs to be extracted from the secret straight in greenhouse, no?

Reply:
gave a more generic way of figuring out obj store config used by compactor